src/04-modality-specific-files/11-functional-near-infrared-spectroscopy.md

```diff
@@ -0,0 +1,400 @@
+
+## fNIRS recording data
+
+{{ MACROS___make_filename_template(datatypes=["nirs"], suffixes=["nirs", "events", "channels", "optodes", "coordsystem"]) }}
+
+Only the Shared Near Infrared Spectroscopy Format ([SNIRF](https://github.com/fNIRS/snirf))
+file specification is supported in BIDS. The SNIRF
+specification supports one or more fNIRS datasets to be stored in a single
+`.snirf` file. However, to be BIDS compatible, each SNIRF file MUST contain
+only a single run. A limited set of fields from the SNIRF specification are
+replicated  in the BIDS specification. This redundancy allows the data to be
+easily parsed by humans and machines that do not have a SNIRF reader at hand,
+which improves findability and tooling development.
+
+Raw fNIRS data in the native format, if different from SNIRF, can also
+be stored in the [`/sourcedata` directory](../02-common-principles.md#source-vs-raw-vs-derived-data)
+along with code to convert the data to
+SNIRF in the [`/code` directory](../02-common-principles.md#storage-of-derived-datasets).
+The unprocessed raw data should be stored in
+the manufacturer's format before any additional processing or conversion is applied.
+Retaining the native file format is especially valuable in a case when conversion elicits the
+loss of crucial metadata unique to specific manufacturers and fNIRS systems.
+
+### Terminology
+
+For proper documentation of fNIRS recording metadata, it is important
+to understand the difference between a Source, Detector, and Channel as these
+are defined differently to other modalities, such as EEG. The following definitions
+apply in this document:
+
+-   Source - A light emitting device, sometimes called a transmitter.
+
+-   Detector - A photoelectric transducer, sometimes called a receiver.
+
+-   Optode - Refers to either a source or detector.
+
+-   Channel - A paired coupling of a source and a detector with one specific wavelength of light.
+    It is common for a single Source-Detector pair to result in two or more channels
+    with different wavelengths.
+
+### Sidecar JSON (`*_nirs.json`)
+
+It is common within the fNIRS community for researchers to build their own caps
+and optode holders to position their sources and detectors, or for optodes to
+be directly attached to the scalp with adhesive. To facilitate description of
+the wide variety of possible configurations, several fields are RECOMMENDED within
+the `*_nirs.json` file.
+To clarify the usage and interaction of these fields, the following examples are provided:
+
+-   If a commercial cap such as EasyCap actiCAP 64 Ch Standard-2 was used:
+    ```JSON
+    "CapManufacturer": "EasyCap",
+    "CapManufacturersModelName": "actiCAP 64 Ch Standard-2",
+    "NIRSPlacementScheme": "n/a"
+    ```
+
+-   If an EasyCap was used but with custom positions,
+    as may be done by cutting custom holes in the cap,
+    was used:
+    ```JSON
+    "CapManufacturer": "EasyCap",
+    "CapManufacturersModelName": "custom",
```

Comment:
Maybe our example on the finger/foot task can help here. See fore example [this file](https://github.com/robertoostenveld/bids-examples/commit/98bb25cd26e67f7de4d2f0770678695b842fe6eb#diff-e8e80f34d6a56741d95def140333e60bfdeb3ba02aa351d6a34506a8043dfe3c). 
We set the CapManufacturersModelName to "headcap with print, customized (L)", but I like your idea to add a field CapModelCustomized. It would then become something like this:
	"CapManufacturer": "Artinis Medical Systems",
	"CapManufacturersModelName": "headcap with print (L)",
        "CapModelCustomized": true,
        
Regarding the NIRSPlacementSchema we solved it this way: 
	"NIRSPlacementScheme": "see optode_layout.pdf: 2 groups over the left and right dlPFC, 2 groups over the left and right PPC, 1 group over the left M1 and PMC",
We then added the optode_layout.pdf in the .bidsignore file. (I misunderstood this first and placed the optode_layout.pdf in .bidsignore folder instead of adding the file in the main folder and writing the filename in the .bidsignore text file, but will update the correct version soon).

Reply:
I am not really enthusiastic about the boolean `CapModelCustomized` metadata field, nor about `CapManufacturersModelName` having to be selected from a list of restricted keywords. There will be other models in the future and we don't want to update the spec. And writing that it is custom does yet not give sufficient details (e.g. how was it modified?). The proposal is not optimal from human readability point of view, and although it is machine readable, the machine (software) cannot make much sense of it. 

I would prefer something like this 

`CapManufacturersModelName`:  `headcap with print, size L, it was modified by adding holes for the optodes  according to the NIRSPlacementScheme and optode_layout.pdf `

Are there actually well-defined NIRS cap model names given by the manufacturers? For EEG we for example have the [ECI types](https://www.electrocap.nl/index.php?page=caps&cat=1) and  the [Easycap types](https://www.easycap.de/wp-content/uploads/2019/10/Partial_PRODUCT-OVERVIEW-08-2019.pdf) which are detailed and presumably unique codes, not just a "head cap with print, sized L".

Reply:
No, there are not yet any well defined cap models. We would like to come up with 3ish standards, but this will take a few years to achieve. For now, your proposed solution sounds good and seems to me that it will be future proof to these 3ish standards we hope to achieve.

Reply:
Thank you all for the detailed discussion. This has been one of the tougher aspects of the specification to get right, and I think we are almost there. Based on the comments above, how about...

| Key name                  | Requirement Level | Data type                  | Description                                                                                                                                                                                                                                                                                                                                                                                                                  |
|---------------------------|-------------------|----------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| CapManufacturer           | RECOMMENDED       | string                     | Name of the cap manufacturer (for example, "EasyCap"). If a custom-made cap is used then the string `custom` MUST be used. If no cap was used, such as with optodes that are directly taped to the scalp, then the string  `none`  MUST be used and the NIRSPlacementScheme field MAY be used to specify the optode placement.                                                                                                   |
| CapManufacturersModelName | RECOMMENDED       | string                     | Manufacturer's designation of the cap model (for example, "actiCAP 64 Ch Standard-2"). If there is no official model number then a description may be provided (for example, Headband with print (S-M)). If a cap from a standard manufacturer was modified, then the field SHOULD be used to describe the cap and modification. If no cap was used, then the CapManufacturer field MUST be none and this field MUST be n/a. |
| NIRSPlacementScheme       | RECOMMENDED       | string or array of strings | Placement scheme of NIRS optodes. Either the name of a standardized placement system (for example, "10-20"), an array of standardized position names (for example, ["Cz", "Pz"]), or description of placement should be used. Reference to further documentation or figures can be placed here. If a standard cap was used, then it should be specified in CapManufacturer and CapManufacturersModelName and this field MAY be used to describe the placement scheme. |

And then we would have the following examples...

-   If a commercial cap such as EasyCap actiCAP 64 Ch Standard-2 was used:
    ```JSON
    "CapManufacturer": "EasyCap",
    "CapManufacturersModelName": "actiCAP 64 Ch Standard-2",
    "NIRSPlacementScheme": "10-20"
    ```

-   If an Artinis Medical Systems cap with custom positions,
    as may be done by cutting custom holes in the cap,
    was used:
    ```JSON
    "CapManufacturer": "Artinis Medical Systems",
    "CapManufacturersModelName": "headcap with print, size L, it was modified by adding holes for the optodes according to the NIRSPlacementScheme and optode_layout.pdf",
    "NIRSPlacementScheme": "see optode_layout.pdf: 2 groups over the left and right dlPFC, 2 groups over the left and right PPC, 1 group over the left M1 and PMC"
    ```

-   If a completely custom cap was knitted:
    ```JSON
    "CapManufacturer": "custom",
    "CapManufacturersModelName": "custom knitted cap with holes for optodes according to the NIRSPlacementScheme and optode_knitted_layout.jpg",
    "NIRSPlacementScheme": "see optode_knitted_layout.jpg: 2 groups over the left and right dlPFC, 2 groups over the left and right PPC"
    ```

-   If no cap was used and optodes were taped to the scalp
    at positions Cz, C1 and C2:
    ```JSON
    "CapManufacturer": "none",
    "CapManufacturersModelName": "n/a",
    "NIRSPlacementScheme": ["Cz", "C1", "C2"],
    ```


> The proposal is not optimal from human readability point of view, and although it is machine readable, the machine (software) cannot make much sense of it.

The proposal is no better at being machine readable, but is it more human readable? Alternative proposals are welcome.

Reply:
I like this suggestion -- some comments:

- Can we standardize the `optode_layout.pdf` to behave something like `_photo.jpg`? https://bids-specification.readthedocs.io/en/latest/04-modality-specific-files/03-electroencephalography.html#landmark-photos-_photojpg ... or do you all judge that this would overload `_photo` by too much? @robertoostenveld 
- We should explicitly declare `"custom"` and `"none"` as reserved keywords for `CapManufacturer` to be used for the cases outlined above. I wouldn't want to be ambiguous whether people may also use `"n/a"` in those cases.

Reply:
I don't like for `CapManufacturer` that _"If a custom-made cap is used then the string "custom" MUST be used."_ I would say that the manufacturer is nearly always well defined. It can be the name of a company, the name of a university, the name of a technical support group within a department, or the name of the tailor. E.g. https://www.ru.nl/socialsciences/technicalsupportgroup/ or http://kinh-do.nl would be well-defined places where I might have a custom cap made in Nijmegen, and I might specify "RU socsci TSG"  or "Kinh-Do". Or if it is not an organization, the person doing the sewing or knitting can be specified. The cap being custom made-to-order or being a confection off-the-shelf cap needs to be represented in the `CapManufacturersModelName` field.

So I propose

```
"CapManufacturer": "the mother of the first author",
"CapManufacturersModelName": "custom knitted cap with holes for optodes according to the NIRSPlacementScheme and optode_knitted_layout.jpg",
"NIRSPlacementScheme": "see optode_knitted_layout.jpg: 2 groups over the left and right dlPFC, 2 groups over the left and right PPC"
```

Re `CapManufacturersModelName`: I propose to scrap "standard" in the phrase  "...standard manufacturer...". The manufacturer is not standard, but the cap might be a standard model for the manufacturer (like a 501 for Levis).

Reply:
I have address this in ad74ef2b1383901c5d1780c32273c1bd2c864d15 and e678b15379f0d653d262ba9661a9a334374791f7.

This is a subtle but important discussion. Please review my changes and let me know if I missed some details. Thank you all for your attention to detail and input above.